operator/pkg/validate/common.go

```diff
@@ -284,6 +284,11 @@ func UnmarshalIOP(iopYAML string) (*v1alpha1.IstioOperator, error) {
 		un.SetCreationTimestamp(meta_v1.Time{}) // UnmarshalIstioOperator chokes on these
 		iopYAML = util.ToYAML(un)
 	}
+	// TODO: find a better way to validate all the invalid values in IOP YAML
+	// before it is unmarshal into IstioOperator
```

Comment:
Not sure if the code has changed or I am just repeating myself, but I still don't think this is right.

First, this method is extremely inefficient. First we take a yaml string, then marshal to a map, then back to yaml, then marshal to a map again, then marshal to IstioOperator. This feels very wrong and contributes to massive memory allocations (it may seem like speed doesn't matter much since its a one-shot CLI tool, and it doesn't, but the amount of allocations we do is so high it makes our unit tests noticeably slow)

Second, why is revision special? Don't we have the same problem with every other field?

Reply:
We had problems in the past with IOPs with creation timestamps.  (e.g. _istioctl/pkg/install/pre-check.go_, _istioctl/pkg/verifier/verifier.go_ and already in _operator/pkg/validate/common.go_.). However, I have retested by removing the clearing of creation time from `istioctl x precheck` and I see no problems.  Let's stop introducing the clearing of Time fields until we can identify a way to reproduce.